### PR TITLE
[FL-3060] New MFC Bruteforce animation

### PR DESCRIPTION
**Pull request by Astrrra**
# What's new

- New MFC bruteforce screen that shows more data on how the process is going
- New "status bar with text" element
- New XOR color for inverting existing screen contents

# Verification 

- Read a Mifare Classic card, check that the new screen is shown and that the keys are advanced in multiples of 10.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/main/nfc/scenes/nfc_scene_mf_classic_dict_attack.c [hunk 1]
```diff
@@ -53,22 +53,22 @@ static void nfc_scene_mf_classic_dict_attack_prepare_view(Nfc* nfc, DictAttackSt
     // Setup view
     if(state == DictAttackStateUserDictInProgress) {
         worker_state = NfcWorkerStateMfClassicDictAttack;
-        dict_attack_set_header(nfc->dict_attack, "Mf Classic User Dict.");
+        dict_attack_set_header(nfc->dict_attack, "MF Classic User Dictuionary");
```
Comment (sodoku) [inline applications/main/nfc/scenes/nfc_scene_mf_classic_dict_attack.c]: Small typo

Comment (Astrrra) [reply applications/main/nfc/scenes/nfc_scene_mf_classic_dict_attack.c]: Thanks!

File: applications/main/nfc/views/dict_attack.c [hunk 2]
```diff
@@ -36,10 +38,20 @@ static void dict_attack_draw_callback(Canvas* canvas, void* model) {
...
+        if(m->is_key_attack) {
+            snprintf(
+                draw_str,
+                sizeof(draw_str),
+                "Reuse key check for sector: %d",
+                m->key_attack_current_sector);
+            canvas_draw_str_aligned(canvas, 0, 10, AlignLeft, AlignTop, draw_str);
+        } else {
+            snprintf(draw_str, sizeof(draw_str), "Unlocking sector: %d", m->sector_current);
+            canvas_draw_str_aligned(canvas, 0, 10, AlignLeft, AlignTop, draw_str);
```
Comment (gornekich) [inline applications/main/nfc/views/dict_attack.c]: repeats line 50

File: applications/main/nfc/views/dict_attack.c [hunk 3]
```diff
@@ -235,3 +249,28 @@ void dict_attack_inc_current_dict_key(DictAttack* dict_attack, uint16_t keys_tri
...
 }
+
+void dict_attack_set_key_attack(DictAttack* dict_attack, bool is_key_attack) {
+    furi_assert(dict_attack);
+    with_view_model(
+        dict_attack->view,
+        DictAttackViewModel * model,
+        {
+            model->is_key_attack = is_key_attack;
+            model->key_attack_current_sector = 1;
```
Comment (gornekich) [inline applications/main/nfc/views/dict_attack.c]: You start key attack not always from 1 sector

File: applications/services/gui/elements.c [hunk 4]
```diff
@@ -41,6 +41,31 @@ void elements_progress_bar(Canvas* canvas, uint8_t x, uint8_t y, uint8_t width,
...
 
+void elements_progress_bar_with_text(
+    Canvas* canvas,
+    uint8_t x,
+    uint8_t y,
+    uint8_t width,
+    float progress,
+    const char* text) {
+    furi_assert(canvas);
+    furi_assert((progress >= 0) && (progress <= 1.0));
```
Comment (gornekich) [inline applications/services/gui/elements.c]: 0.0f, 1.0f

**Review by gornekich — DISMISSED** [review]
Looks awesome!
The major issue with key attack current sector.

File: applications/main/nfc/views/dict_attack.c (hunk 3 above)
Comment (Astrrra) [reply applications/main/nfc/views/dict_attack.c]: Fixed

**Review by skotopes — APPROVED** [review]

**Pull request merged**